cpp/src/gandiva/precompiled/timestamp_arithmetic.cc

```diff
@@ -239,4 +290,18 @@ ADD_TIMESTAMP_TO_INT64_FIXED_UNITS(date64, add, MILLIS_IN_DAY)
 ADD_TIMESTAMP_TO_INT64_FIXED_UNITS(timestamp, date_add, MILLIS_IN_DAY)
 ADD_TIMESTAMP_TO_INT64_FIXED_UNITS(timestamp, add, MILLIS_IN_DAY)
 
+// add a time interval to timestamp
+ADD_DAY_TIME_INTERVAL_TO_DATE_TYPES(date64, add, MILLIS_IN_DAY)
```

Comment:
2nd, 3rd argument not required.
Also you can further create a single macro for all add/sub day_time_interval/year_month_interval

Reply:
fixed